### PR TITLE
Cleanup & QOL

### PR DESCRIPTION
**Pull request by zoller27osu**
FindMethod with argCount -> FindMethodUnsafe (attempts to still use FindMethod result in descriptive compile-time error)
Replaced all class_get_name + class_get_namespace with ClassStandardName
Moved SAFE_ABORT to utils.h
Tested and tweaked timings & aborts

### CODE REVIEW

File: shared/customui/customui.cpp [hunk 1]
```diff
@@ -39,132 +38,114 @@ namespace CustomUI
...
-            if (!RunMethod(&assetName, allObjects->values[i], "get_name"))
-            {
+            if (!RunMethod(&assetName, allObjects->values[i], "get_name")) {
                 // EXCEPTION
                 log(ERROR, "TextObject::create: Failed to run get_name of assetName");
                 return false;
             }
-            if (strcmp(to_utf8(csstrtostr(assetName)).c_str(), "Teko-Medium SDF No Glow") == 0)
-            {
+            if (strcmp(to_utf8(csstrtostr(assetName)).c_str(), "Teko-Medium SDF No Glow") == 0) {
```
Comment (sc2ad) [inline shared/customui/customui.cpp]: Should transition away from these methods and instead use the std::string equals operator (given that this is a string_view, it should behave as such)

File: shared/utils/il2cpp-functions.cpp [hunk 2]
```diff
@@ -564,6 +559,6 @@ void il2cpp_functions::Init() {
 
     dlclose(imagehandle);
     initialized = true;
+    usleep(5000L);  // 0.005s
```
Comment (sc2ad) [inline shared/utils/il2cpp-functions.cpp:555]: Do we need to _always_ delay execution in `il2cpp_functions::Init`? It seems like this could be avoided. Why was this required?

File: shared/utils/il2cpp-utils.cpp [hunk 3]
```diff
@@ -625,7 +632,7 @@ namespace il2cpp_utils {
 
         bool methodInit = false;
         if (klass->name) {
-            // note: unless vm/Class.cpp is wrong, Class::Init always returns true
+            // Note: unless vm/Class.cpp is wrong, Class::Init always returns true
```
Comment (sc2ad) [inline shared/utils/il2cpp-utils.cpp:594]: ah, man. How I love il2cpp!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

File: shared/utils/utils.h [hunk 4]
```diff
@@ -23,6 +23,14 @@ template <typename Container> struct is_vector : std::false_type { };
 template <typename... Ts> struct is_vector<std::vector<Ts...> > : std::true_type { };
 // TODO: figure out how to write an is_vector_v that compiles properly?
 
+#define MACRO_WRAP(x) do { \
+    x; \
+} while(0)
+
+#define CRASH_IF_NOT(arg) MACRO_WRAP( \
+    if (!(arg)) SAFE_ABORT(); \
```
Comment (sc2ad) [inline shared/utils/utils.h]: Shouldn't actually need to wrap a one line'd if statement, since it'll still functionally work when used _as_ a one liner (the if will just nest to the next line as intended anyways).
Typically, wraps are for multi-line defines.

Comment (sc2ad) [reply shared/utils/utils.h]: In addition, this new method should be used as a replacement in _many_ locations, or a similar macro should be made for simply returning false with a critical log. That would simplify a lot of the redundancy/boilerplate-y-ness of customui

**Review by sc2ad — CHANGES_REQUESTED** [review]
Left a few comments I'd like answered before merging this. In general, a much better QOL commit than bs-hook has had for quite some time. Thanks!

File: shared/customui/customui.cpp (hunk 1 above)
Comment (zoller27osu) [reply shared/customui/customui.cpp]: Ah, good catch

File: shared/utils/il2cpp-functions.cpp (hunk 2 above)
Comment (zoller27osu) [reply shared/utils/il2cpp-functions.cpp:555]: These pauses are to allow the logging to finish. I still don't know exactly when/why it happens, but without these pauses (and sometimes with them) some logging messages just get clobbered out of existence.
I've read that there's some sort of ring buffer, but my current hypothesis is that once the buffer is full, subsequent logs just won't work. So after we log lots, we sleep.

These pauses are honestly quite minuscule compared to what they used to be, though.

File: shared/utils/utils.h (hunk 4 above)
Comment (zoller27osu) [reply shared/utils/utils.h]: `if (blah) if (blah) blah;` compiles fine? ok

Comment (zoller27osu) [reply shared/utils/utils.h]: nah fam sounds dangerous https://stackoverflow.com/questions/11734604/if-block-without-curly-braces-makes-subsequent-else-if-nested

Comment (zoller27osu) [reply shared/utils/utils.h]: > a similar macro should be made for simply returning false with a critical log. That would simplify a lot of the redundancy/boilerplate-y-ness of customui

done via RET(_0)_UNLESS

File: shared/utils/il2cpp-functions.cpp (hunk 2 above)
Comment (zoller27osu) [reply shared/utils/il2cpp-functions.cpp:555]: The first pause is for all the dlsym logs.
The 2nd pause is for the logs generated by all of the Instruction parsing that xref tracing does.

**Review by sc2ad — APPROVED** [review]
All good to go!
